src/recordlinker/linking/link.py

```diff
@@ -144,7 +144,11 @@ def link_record_against_mpi(
             # block on the pii_record and the algorithm's blocking criteria, then
             # iterate over the patients, grouping them by person
             with TRACER.start_as_current_span("link.block"):
-                patients = mpi_service.get_block_data(session, record, algorithm_pass)
+                # get all candidate Patient records identified in blocking
+                candidate_patients = mpi_service.get_block_data(session, record, algorithm_pass)
+                # fetch all remaining Patient records within the Person clusters identified in
+                # blocking for Belongingess Ratio calculation
+                patients = mpi_service.fetch_person_records(session, candidate_patients)
```

Comment:
With this implementation, you will run an extra DB query for every blocking pass.  Have you tested the performance implications of adding these?

Reply:
Great point, performance implications with an additional DB query was an oversight of mine.

Just pushed changes to give another pass. I looked into [Lazy Loading](https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html#lazy-loading) with SQLAlchemy ORM, but wondering if returning the query expression for `candidate_patients` in `get_block_data()` and passing that to `get_person_records()` where the query is executed suffices.